src/Microsoft.TestPlatform.Common/ExtensionDecorators/SerializeTestRunDecorator.cs

```diff
@@ -0,0 +1,137 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Xml.Linq;
+
+using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel.Adapter;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel.Logging;
+
+namespace Microsoft.VisualStudio.TestPlatform.Common.ExtensionDecorators;
+
+internal class SerializeTestRunDecorator : ITestExecutor, ITestExecutor2, IDisposable
```

Comment:
I would use something else than `Serialize` to avoid naming confusion for devs. Maybe `SerialTestRunDectorator`?

Reply:
Lol dev should know what serialize is in the context of lock :), I'll rename.